### PR TITLE
Remove msgpack and add basic jsonschema support

### PR DESCRIPTION
**Pull request by paulbovbel**
Nothing fancy here, particularly in how jsonschema is used. Extended the PackedJson wrapper to support providing a schema when the message is populated with data.

Is there something more we'd like to see from the API here? I didn't want to go too crazy creating publisher/subscriber subclasses.

Additionally, C++ makes it much harder to 'wrap' things, and the only hook we have is to allow a nlohmann::json type to masquerade as a json_msgs/Json message, which was already done - there's no place to inject a schema at all. Probably the best path there is to have a 'best practice' of using a library like https://github.com/pboettch/json-schema-validator. Should we 'vendor' the library here just like nlohmann:json? Seems premature.

### CODE REVIEW

**Review by garyservin — APPROVED** [review]

File: json_transport/src/json_transport/json_transport.py [hunk 1]
```diff
@@ -23,29 +23,32 @@
...
-    return msgpack.unpackb(message.bytes, encoding="utf-8")
+    return json.loads(message.json)
 
 
 class PackedJson(json_msg.Json):
 
-    def __init__(self, data=None):
+    def __init__(self, data=None, schema=None):
+        if schema:
+            validate(instance=data, schema=schema)
```
Comment (ablakey) [inline json_transport/src/json_transport/json_transport.py:44]: LGTM. The one thing to keep in mind (probably left to the developers): validating high frequency messages against non-trivial schema can be very slow. Though I don't think this changes the "feel" of rospy much given that it already suffers from  "some uses of publish can really be slow".

**Review by ablakey — APPROVED** [review]

**Comment by ablakey** [comment]
Major version bump?

**Comment by paulbovbel** [comment]
> Major version bump?

If/when this gets an upstream release, yeah, for sure.

**Comment by DLu** [comment]
This broke CI for `locus_nav_plugins`: 
 * [Offending line](https://github.com/locusrobotics/locus_nav_plugins/blob/36bddf45317c873ec93d41c46d4163f780f67007/locus_locomotor/locho/test/test_tycho_helper.cpp#L63)

not sure if this is something for @paulbovbel  or @tappan-at-git  to fix.

**Comment by paulbovbel** [comment]
I'll fix it, thanks!

**Pull request merged**